boot/kernel_os.go

```diff
@@ -147,6 +148,16 @@ func SetNextBoot(s *snap.Info) error {
 		return nil
 	}
 
+	// With bases we need to check if this is a base we want to
+	// boot from, i.e. if the snap has the same name as the one
+	// we used for booting.
+	snapNameGoodBoot := strings.SplitN(m[goodBoot], "_", 2)[0]
+	if m[goodBoot] != "" && snapNameGoodBoot != s.Name() {
+		// Nothing to do, this is just a base snap but not
+		// the base name we used for booting.
+		return nil
+	}
```

Comment:
mmh, this check in this form feels a bit of a hack, wondering if we can do better

Reply:
notice that devicestate already sets various helpers for use in snapstate,   it could also set some helper to know what is the system model base

Reply:
I agree, using the devicestate.Model() is best here. I can tweak this. One reason it is done the way it is right now is that we don't have a core18 model so in order to test refreshes from core18 -> core18 this is the most practical way right now. But I will look into make this better.